### ISSUE TITLE
Neither Gazebo Classic nor jmavsim will work on Mac M1 arm

### ISSUE THREAD

**Issue by DonLakeFlyer** [PX4/PX4-Autopilot#23013]
### Describe the bug

I followed all the setup instructions including using an iTerm install which is running in Rosetta. This leaves me pretty hosed with respect to doing QGC testing.

Gazebo classic:
```
don@Donalds-Mac-Pro PX4-Autopilot % brew install px4-sim-gazebo
==> Downloading https://formulae.brew.sh/api/formula.jws.json

==> Downloading https://formulae.brew.sh/api/cask.jws.json

==> Fetching dependencies for px4/px4/px4-sim-gazebo: osrf/simulation/gazebo11
==> Fetching osrf/simulation/gazebo11
==> Downloading https://github.com/gazebosim/gazebo-classic/commit/3a9efee7ccdf552cbf22188131782f242f6c0542.patch?full_index=1
Already downloaded: /Users/don/Library/Caches/Homebrew/downloads/740e15a70c717de3cfc904342743e25cd1d228034f3573d51f23d6ad6093e0be--3a9efee7ccdf552cbf22188131782f242f6c0542.patch
==> Downloading https://gist.githubusercontent.com/scpeters/9199370/raw/afe595587e38737c537124a3652db99de026c272/brew_python_fix.patch
Already downloaded: /Users/don/Library/Caches/Homebrew/downloads/12171d5787bf5709495604c1591fd598a9cffc3a055eb509bea588d33019cabc--brew_python_fix.patch
==> Downloading https://osrf-distributions.s3.amazonaws.com/gazebo/releases/gazebo-11.14.0.tar.bz2
Already downloaded: /Users/don/Library/Caches/Homebrew/downloads/886d578ff7aa32ac5da8b63cc2ff79f24be5c9aff5092de102f6c6fd242f8b3d--gazebo-11.14.0.tar.bz2
==> Fetching px4/px4/px4-sim-gazebo
==> Downloading https://raw.githubusercontent.com/PX4/PX4-Autopilot/master/Tools/px4.py
Already downloaded: /Users/don/Library/Caches/Homebrew/downloads/be4aff9b473b8b0c6bae0ff08a6611885de9d56ee3a477c05453bb6bbcaab246--px4.py
==> Installing px4-sim-gazebo from px4/px4
==> Installing dependencies for px4/px4/px4-sim-gazebo: osrf/simulation/gazebo11
==> Installing px4/px4/px4-sim-gazebo dependency: osrf/simulation/gazebo11
==> Patching
==> Applying 3a9efee7ccdf552cbf22188131782f242f6c0542.patch
patching file 'gazebo/gui/qgv/QGVSubGraph.cpp'
==> Applying brew_python_fix.patch
patching file 'deps/opende/CMakeLists.txt'
==> cmake .. -DQWT_WIN_INCLUDE_DIR=/opt/homebrew/opt/qwt-qt5/lib/qwt.framework/Headers -DQWT_WIN_LIBRARY_DIR=/opt/homebrew/opt/qwt-qt5/lib/qwt.framework -DCMAKE_INSTALL_RPATH=@loader_path/../lib
==> make install
Last 15 lines from /Users/don/Library/Logs/Homebrew/gazebo11/02.make:
cd /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende && /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DDART_ACTIVE_LOG_LEVEL=2 -DDART_HAVE_spdlog=1 -DFMT_HEADER_ONLY=1 -DFMT_SHARED -DHAVE_OPENGL -DIGN_PROFILER_ENABLE=0 -DLIBBULLET_VERSION=3.25 -DLIBBULLET_VERSION_GT_282 -DMAC_OS_X_VERSION=1080 -DSPDLOG_FMT_EXTERNAL -DTINYXML2_IMPORT -D_FILE_OFFSET_BITS=64 -Dgazebo_ode_EXPORTS -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/test/gtest/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0 -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/ou/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/OPCODE -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/GIMPACT/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/threadpool -I/opt/homebrew/Cellar/tinyxml2/10.0.0/include -I/opt/homebrew/Cellar/ffmpeg/7.0/include -I/opt/homebrew/include/ignition/msgs5 -I/opt/homebrew/include/ignition/transport8 -I/opt/homebrew/include/ignition/common3 -I/opt/homebrew/include/ignition/fuel_tools4 -I/opt/homebrew/Cellar/libccd/2.1_1/include -I/opt/homebrew/Cellar/bullet/3.25/include/bullet -I/opt/homebrew/Cellar/bullet/3.25/include -isystem /opt/homebrew/include/ignition/math6 -isystem /opt/homebrew/include/eigen3 -isystem /opt/homebrew/include/bullet -isystem /opt/homebrew/lib/urdfdom/cmake/../../../include/urdfdom -isystem /opt/homebrew/lib/urdfdom_headers/cmake/../../../include/urdfdom_headers -stdlib=libc++  -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal -Wshadow -Wswitch-default -Wmissing-include-dirs -pedantic  -DNDEBUG -DdNODEBUG -DdDOUBLE -DHAVE_CONFIG_H -DPIC  -s -O3 -DNDEBUG  -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -fPIC -MD -MT deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/dhinge.cpp.o -MF CMakeFiles/gazebo_ode.dir/src/joints/dhinge.cpp.o.d -o CMakeFiles/gazebo_ode.dir/src/joints/dhinge.cpp.o -c /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/src/joints/dhinge.cpp
[ 14%] Building CXX object deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/dball.cpp.o
cd /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende && /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DDART_ACTIVE_LOG_LEVEL=2 -DDART_HAVE_spdlog=1 -DFMT_HEADER_ONLY=1 -DFMT_SHARED -DHAVE_OPENGL -DIGN_PROFILER_ENABLE=0 -DLIBBULLET_VERSION=3.25 -DLIBBULLET_VERSION_GT_282 -DMAC_OS_X_VERSION=1080 -DSPDLOG_FMT_EXTERNAL -DTINYXML2_IMPORT -D_FILE_OFFSET_BITS=64 -Dgazebo_ode_EXPORTS -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/test/gtest/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0 -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/ou/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/OPCODE -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/GIMPACT/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/threadpool -I/opt/homebrew/Cellar/tinyxml2/10.0.0/include -I/opt/homebrew/Cellar/ffmpeg/7.0/include -I/opt/homebrew/include/ignition/msgs5 -I/opt/homebrew/include/ignition/transport8 -I/opt/homebrew/include/ignition/common3 -I/opt/homebrew/include/ignition/fuel_tools4 -I/opt/homebrew/Cellar/libccd/2.1_1/include -I/opt/homebrew/Cellar/bullet/3.25/include/bullet -I/opt/homebrew/Cellar/bullet/3.25/include -isystem /opt/homebrew/include/ignition/math6 -isystem /opt/homebrew/include/eigen3 -isystem /opt/homebrew/include/bullet -isystem /opt/homebrew/lib/urdfdom/cmake/../../../include/urdfdom -isystem /opt/homebrew/lib/urdfdom_headers/cmake/../../../include/urdfdom_headers -stdlib=libc++  -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal -Wshadow -Wswitch-default -Wmissing-include-dirs -pedantic  -DNDEBUG -DdNODEBUG -DdDOUBLE -DHAVE_CONFIG_H -DPIC  -s -O3 -DNDEBUG  -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -fPIC -MD -MT deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/dball.cpp.o -MF CMakeFiles/gazebo_ode.dir/src/joints/dball.cpp.o.d -o CMakeFiles/gazebo_ode.dir/src/joints/dball.cpp.o -c /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/src/joints/dball.cpp
[ 14%] Building CXX object deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/gearbox.cpp.o
cd /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende && /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DDART_ACTIVE_LOG_LEVEL=2 -DDART_HAVE_spdlog=1 -DFMT_HEADER_ONLY=1 -DFMT_SHARED -DHAVE_OPENGL -DIGN_PROFILER_ENABLE=0 -DLIBBULLET_VERSION=3.25 -DLIBBULLET_VERSION_GT_282 -DMAC_OS_X_VERSION=1080 -DSPDLOG_FMT_EXTERNAL -DTINYXML2_IMPORT -D_FILE_OFFSET_BITS=64 -Dgazebo_ode_EXPORTS -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/test/gtest/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0 -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/ou/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/OPCODE -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/GIMPACT/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/threadpool -I/opt/homebrew/Cellar/tinyxml2/10.0.0/include -I/opt/homebrew/Cellar/ffmpeg/7.0/include -I/opt/homebrew/include/ignition/msgs5 -I/opt/homebrew/include/ignition/transport8 -I/opt/homebrew/include/ignition/common3 -I/opt/homebrew/include/ignition/fuel_tools4 -I/opt/homebrew/Cellar/libccd/2.1_1/include -I/opt/homebrew/Cellar/bullet/3.25/include/bullet -I/opt/homebrew/Cellar/bullet/3.25/include -isystem /opt/homebrew/include/ignition/math6 -isystem /opt/homebrew/include/eigen3 -isystem /opt/homebrew/include/bullet -isystem /opt/homebrew/lib/urdfdom/cmake/../../../include/urdfdom -isystem /opt/homebrew/lib/urdfdom_headers/cmake/../../../include/urdfdom_headers -stdlib=libc++  -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal -Wshadow -Wswitch-default -Wmissing-include-dirs -pedantic  -DNDEBUG -DdNODEBUG -DdDOUBLE -DHAVE_CONFIG_H -DPIC  -s -O3 -DNDEBUG  -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -fPIC -MD -MT deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/gearbox.cpp.o -MF CMakeFiles/gazebo_ode.dir/src/joints/gearbox.cpp.o.d -o CMakeFiles/gazebo_ode.dir/src/joints/gearbox.cpp.o -c /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/src/joints/gearbox.cpp
[ 14%] Building CXX object deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/slider.cpp.o
cd /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende && /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DDART_ACTIVE_LOG_LEVEL=2 -DDART_HAVE_spdlog=1 -DFMT_HEADER_ONLY=1 -DFMT_SHARED -DHAVE_OPENGL -DIGN_PROFILER_ENABLE=0 -DLIBBULLET_VERSION=3.25 -DLIBBULLET_VERSION_GT_282 -DMAC_OS_X_VERSION=1080 -DSPDLOG_FMT_EXTERNAL -DTINYXML2_IMPORT -D_FILE_OFFSET_BITS=64 -Dgazebo_ode_EXPORTS -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/test/gtest/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0 -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/ou/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/OPCODE -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/GIMPACT/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/threadpool -I/opt/homebrew/Cellar/tinyxml2/10.0.0/include -I/opt/homebrew/Cellar/ffmpeg/7.0/include -I/opt/homebrew/include/ignition/msgs5 -I/opt/homebrew/include/ignition/transport8 -I/opt/homebrew/include/ignition/common3 -I/opt/homebrew/include/ignition/fuel_tools4 -I/opt/homebrew/Cellar/libccd/2.1_1/include -I/opt/homebrew/Cellar/bullet/3.25/include/bullet -I/opt/homebrew/Cellar/bullet/3.25/include -isystem /opt/homebrew/include/ignition/math6 -isystem /opt/homebrew/include/eigen3 -isystem /opt/homebrew/include/bullet -isystem /opt/homebrew/lib/urdfdom/cmake/../../../include/urdfdom -isystem /opt/homebrew/lib/urdfdom_headers/cmake/../../../include/urdfdom_headers -stdlib=libc++  -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal -Wshadow -Wswitch-default -Wmissing-include-dirs -pedantic  -DNDEBUG -DdNODEBUG -DdDOUBLE -DHAVE_CONFIG_H -DPIC  -s -O3 -DNDEBUG  -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -fPIC -MD -MT deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/slider.cpp.o -MF CMakeFiles/gazebo_ode.dir/src/joints/slider.cpp.o.d -o CMakeFiles/gazebo_ode.dir/src/joints/slider.cpp.o -c /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/src/joints/slider.cpp
[ 14%] Building CXX object deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/universal.cpp.o
cd /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende && /opt/homebrew/Library/Homebrew/shims/mac/super/clang++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_NO_LIB -DBOOST_REGEX_DYN_LINK -DBOOST_REGEX_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DDART_ACTIVE_LOG_LEVEL=2 -DDART_HAVE_spdlog=1 -DFMT_HEADER_ONLY=1 -DFMT_SHARED -DHAVE_OPENGL -DIGN_PROFILER_ENABLE=0 -DLIBBULLET_VERSION=3.25 -DLIBBULLET_VERSION_GT_282 -DMAC_OS_X_VERSION=1080 -DSPDLOG_FMT_EXTERNAL -DTINYXML2_IMPORT -D_FILE_OFFSET_BITS=64 -Dgazebo_ode_EXPORTS -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/test/gtest/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0 -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/ou/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/OPCODE -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/GIMPACT/include -I/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/threadpool -I/opt/homebrew/Cellar/tinyxml2/10.0.0/include -I/opt/homebrew/Cellar/ffmpeg/7.0/include -I/opt/homebrew/include/ignition/msgs5 -I/opt/homebrew/include/ignition/transport8 -I/opt/homebrew/include/ignition/common3 -I/opt/homebrew/include/ignition/fuel_tools4 -I/opt/homebrew/Cellar/libccd/2.1_1/include -I/opt/homebrew/Cellar/bullet/3.25/include/bullet -I/opt/homebrew/Cellar/bullet/3.25/include -isystem /opt/homebrew/include/ignition/math6 -isystem /opt/homebrew/include/eigen3 -isystem /opt/homebrew/include/bullet -isystem /opt/homebrew/lib/urdfdom/cmake/../../../include/urdfdom -isystem /opt/homebrew/lib/urdfdom_headers/cmake/../../../include/urdfdom_headers -stdlib=libc++  -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal -Wshadow -Wswitch-default -Wmissing-include-dirs -pedantic  -DNDEBUG -DdNODEBUG -DdDOUBLE -DHAVE_CONFIG_H -DPIC  -s -O3 -DNDEBUG  -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -fPIC -MD -MT deps/opende/CMakeFiles/gazebo_ode.dir/src/joints/universal.cpp.o -MF CMakeFiles/gazebo_ode.dir/src/joints/universal.cpp.o.d -o CMakeFiles/gazebo_ode.dir/src/joints/universal.cpp.o -c /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/deps/opende/src/joints/universal.cpp
[ 14%] Linking CXX shared library libgazebo_ode.dylib
cd /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende && /opt/homebrew/Cellar/cmake/3.29.2/bin/cmake -E cmake_link_script CMakeFiles/gazebo_ode.dir/link.txt --verbose=1
/opt/homebrew/Library/Homebrew/shims/mac/super/clang++       -stdlib=libc++  -Wall -Wextra -Wno-long-long -Wno-unused-value -Wfloat-equal -Wshadow -Wswitch-default -Wmissing-include-dirs -pedantic  -DNDEBUG -DdNODEBUG -DdDOUBLE -DHAVE_CONFIG_H -DPIC  -s -O3 -DNDEBUG  -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -dynamiclib -Wl,-headerpad_max_install_names    -Wl,-undefined -Wl,dynamic_lookup  -MF -MT -fno-strict-aliasing -DPIC  -compatibility_version 11.0.0 -current_version 11.14.0 -o libgazebo_ode.11.14.0.dylib -install_name @rpath/libgazebo_ode.11.dylib CMakeFiles/gazebo_ode.dir/src/array.cpp.o CMakeFiles/gazebo_ode.dir/src/box.cpp.o CMakeFiles/gazebo_ode.dir/src/capsule.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_cylinder_box.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_cylinder_plane.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_cylinder_sphere.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_cylinder_trimesh.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_kernel.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_libccd.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_quadtreespace.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_sapspace.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_space.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_transform.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_box.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_ccylinder.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_disabled.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_distance.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_gimpact.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_opcode.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_plane.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_ray.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_sphere.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_trimesh.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_trimesh_trimesh_new.cpp.o CMakeFiles/gazebo_ode.dir/src/collision_util.cpp.o CMakeFiles/gazebo_ode.dir/src/convex.cpp.o CMakeFiles/gazebo_ode.dir/src/cylinder.cpp.o CMakeFiles/gazebo_ode.dir/src/error.cpp.o "CMakeFiles/gazebo_ode.dir/src/export-dif.cpp.o" CMakeFiles/gazebo_ode.dir/src/heightfield.cpp.o CMakeFiles/gazebo_ode.dir/src/io.cpp.o CMakeFiles/gazebo_ode.dir/src/ioh5.cpp.o CMakeFiles/gazebo_ode.dir/src/lcp.cpp.o CMakeFiles/gazebo_ode.dir/src/mass.cpp.o CMakeFiles/gazebo_ode.dir/src/mat.cpp.o CMakeFiles/gazebo_ode.dir/src/matrix.cpp.o CMakeFiles/gazebo_ode.dir/src/memory.cpp.o CMakeFiles/gazebo_ode.dir/src/misc.cpp.o CMakeFiles/gazebo_ode.dir/src/obstack.cpp.o CMakeFiles/gazebo_ode.dir/src/ode.cpp.o CMakeFiles/gazebo_ode.dir/src/odeinit.cpp.o CMakeFiles/gazebo_ode.dir/src/odemath.cpp.o CMakeFiles/gazebo_ode.dir/src/odeou.cpp.o CMakeFiles/gazebo_ode.dir/src/odetls.cpp.o CMakeFiles/gazebo_ode.dir/src/optimize.cpp.o CMakeFiles/gazebo_ode.dir/src/plane.cpp.o CMakeFiles/gazebo_ode.dir/src/quickstep.cpp.o CMakeFiles/gazebo_ode.dir/src/quickstep_cg_lcp.cpp.o CMakeFiles/gazebo_ode.dir/src/quickstep_pgs_lcp.cpp.o CMakeFiles/gazebo_ode.dir/src/quickstep_update_bodies.cpp.o CMakeFiles/gazebo_ode.dir/src/quickstep_util.cpp.o CMakeFiles/gazebo_ode.dir/src/ray.cpp.o CMakeFiles/gazebo_ode.dir/src/robuststep.cpp.o CMakeFiles/gazebo_ode.dir/src/rotation.cpp.o CMakeFiles/gazebo_ode.dir/src/sphere.cpp.o CMakeFiles/gazebo_ode.dir/src/step.cpp.o CMakeFiles/gazebo_ode.dir/src/step_bullet_lemke_wrapper.cpp.o CMakeFiles/gazebo_ode.dir/src/step_bullet_pgs_wrapper.cpp.o CMakeFiles/gazebo_ode.dir/src/step_dart_pgs_wrapper.cpp.o CMakeFiles/gazebo_ode.dir/src/symm.c.o CMakeFiles/gazebo_ode.dir/src/timer.cpp.o CMakeFiles/gazebo_ode.dir/src/util.cpp.o CMakeFiles/gazebo_ode.dir/src/fastdot.c.o CMakeFiles/gazebo_ode.dir/src/fastldlt.c.o CMakeFiles/gazebo_ode.dir/src/fastlsolve.c.o CMakeFiles/gazebo_ode.dir/src/fastltsolve.c.o CMakeFiles/gazebo_ode.dir/src/nextafterf.c.o CMakeFiles/gazebo_ode.dir/src/joints/amotor.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/ball.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/contact.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/fixed.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/hinge2.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/hinge.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/joint.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/lmotor.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/null.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/piston.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/plane2d.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/pr.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/pu.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/screw.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/dhinge.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/dball.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/gearbox.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/slider.cpp.o CMakeFiles/gazebo_ode.dir/src/joints/universal.cpp.o   -L/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/test  -L/opt/homebrew/Cellar/tinyxml2/10.0.0/lib  -L/opt/homebrew/Cellar/ffmpeg/7.0/lib  -L/opt/homebrew/Cellar/libccd/2.1_1/lib  -L/opt/homebrew/Cellar/bullet/3.25/lib  -Wl,-rpath,/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/test -Wl,-rpath,/opt/homebrew/Cellar/tinyxml2/10.0.0/lib -Wl,-rpath,/opt/homebrew/Cellar/ffmpeg/7.0/lib -Wl,-rpath,/opt/homebrew/Cellar/libccd/2.1_1/lib -Wl,-rpath,/opt/homebrew/Cellar/bullet/3.25/lib -Wl,-rpath,/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende/OPCODE -Wl,-rpath,/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende/GIMPACT -Wl,-rpath,/tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende/ou OPCODE/libgazebo_opcode.11.14.0.dylib GIMPACT/libgazebo_gimpact.11.14.0.dylib ou/libgazebo_opende_ou.11.14.0.dylib -lm /opt/homebrew/lib/libboost_thread-mt.dylib /opt/homebrew/lib/libboost_system-mt.dylib /opt/homebrew/lib/libboost_filesystem-mt.dylib /opt/homebrew/lib/libboost_program_options-mt.dylib /opt/homebrew/lib/libboost_regex-mt.dylib /opt/homebrew/lib/libboost_iostreams-mt.dylib /opt/homebrew/lib/libboost_date_time-mt.dylib -lBulletSoftBody -lBulletDynamics -lBulletCollision -lLinearMath /opt/homebrew/lib/libdart-collision-bullet.6.13.2.dylib /opt/homebrew/lib/libdart-collision-ode.6.13.2.dylib /opt/homebrew/lib/libdart-utils-urdf.6.13.2.dylib /opt/homebrew/lib/libboost_atomic-mt.dylib /opt/homebrew/lib/libBulletDynamics.dylib /opt/homebrew/lib/libBulletCollision.dylib /opt/homebrew/lib/libLinearMath.dylib /opt/homebrew/lib/libBulletSoftBody.dylib /opt/homebrew/lib/libode.dylib /opt/homebrew/lib/libdart-utils.6.13.2.dylib /opt/homebrew/lib/libdart.6.13.2.dylib /opt/homebrew/lib/libdart-external-odelcpsolver.6.13.2.dylib /opt/homebrew/lib/libfcl.0.7.0.dylib /opt/homebrew/lib/libccd.2.0.dylib /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/lib/libm.tbd /opt/homebrew/lib/libassimp.dylib /opt/homebrew/lib/libfmt.10.2.1.dylib /opt/homebrew/lib/liboctomap.1.10.0.dylib /opt/homebrew/lib/liboctomath.1.10.0.dylib /opt/homebrew/lib/libtinyxml2.10.0.0.dylib /opt/homebrew/lib/liburdfdom_sensor.dylib /opt/homebrew/lib/liburdfdom_model_state.dylib /opt/homebrew/lib/liburdfdom_model.dylib /opt/homebrew/lib/liburdfdom_world.dylib
cd /tmp/gazebo11-20240412-34395-pqq5q/gazebo-11.14.0/build/deps/opende && /opt/homebrew/Cellar/cmake/3.29.2/bin/cmake -E cmake_symlink_library libgazebo_ode.11.14.0.dylib libgazebo_ode.11.dylib libgazebo_ode.dylib
[ 14%] Built target gazebo_ode
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/osrf/homebrew-simulation/issues

These open issues may also help:
Failed to build gazebo11 on Apple Silicon https://github.com/osrf/homebrew-simulation/issues/2220
```

jmavsim:
```
don@Donalds-Mac-Pro PX4-Autopilot % brew install px4-sim-jmavsim
==> Fetching px4/px4/px4-sim-jmavsim
==> Downloading https://raw.githubusercontent.com/PX4/PX4-Autopilot/master/Tools/px4.py
Already downloaded: /Users/don/Library/Caches/Homebrew/downloads/be4aff9b473b8b0c6bae0ff08a6611885de9d56ee3a477c05453bb6bbcaab246--px4.py
==> Installing px4-sim-jmavsim from px4/px4
==> PX4 jMAVSim simulation installed
🍺  /opt/homebrew/Cellar/px4-sim-jmavsim/1.11.0: 3 files, 12.9KB, built in 1 second
==> Running `brew cleanup px4-sim-jmavsim`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
don@Donalds-Mac-Pro PX4-Autopilot % make -j24 px4_sitl_default jmavsim
[1/2] launching px4 jmavsim_iris (SYS_AUTOSTART=10017)

______  __   __    ___
| ___ \ \ \ / /   /   |
| |_/ /  \ V /   / /| |
|  __/   /   \  / /_| |
| |     / /^\ \ \___  |
\_|     \/   \/     |_/

px4 starting.

INFO  [px4] startup script: /bin/sh etc/init.d-posix/rcS 0
env SYS_AUTOSTART: 10017
INFO  [param] selected parameter default file parameters.bson
INFO  [param] selected parameter backup file parameters_backup.bson
  SYS_AUTOCONFIG: curr: 0 -> new: 1
  SYS_AUTOSTART: curr: 0 -> new: 10017
  CAL_ACC0_ID: curr: 0 -> new: 1310988
  CAL_GYRO0_ID: curr: 0 -> new: 1310988
  CAL_ACC1_ID: curr: 0 -> new: 1310996
  CAL_GYRO1_ID: curr: 0 -> new: 1310996
  CAL_ACC2_ID: curr: 0 -> new: 1311004
  CAL_GYRO2_ID: curr: 0 -> new: 1311004
  CAL_MAG0_ID: curr: 0 -> new: 197388
  CAL_MAG0_PRIO: curr: -1 -> new: 50
  CAL_MAG1_ID: curr: 0 -> new: 197644
  CAL_MAG1_PRIO: curr: -1 -> new: 50
  SENS_BOARD_X_OFF: curr: 0.0000 -> new: 0.0000
  SENS_DPRES_OFF: curr: 0.0000 -> new: 0.0010
INFO  [dataman] data manager file './dataman' size is 7872608 bytes
INFO  [init] jMAVSim simulator
INFO  [simulator_mavlink] using TCP on remote host localhost port 4560
WARN  [simulator_mavlink] Please ensure port 4560 is not blocked by a firewall.
INFO  [simulator_mavlink] Resolved host 'localhost' to address: 127.0.0.1
INFO  [simulator_mavlink] Waiting for simulator to accept connection on TCP port 4560
Buildfile: /Users/don/repos/PX4-Autopilot/Tools/simulation/jmavsim/jMAVSim/build.xml

make_dirs:

compile:

create_run_jar:

copy_res:

BUILD SUCCESSFUL
Total time: 0 seconds
Options parsed, starting Sim.
Starting GUI...
Exception in thread "main" java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.eclipse.jdt.internal.jarinjarloader.JarRsrcLoader.main(JarRsrcLoader.java:61)
Caused by: java.lang.UnsatisfiedLinkError: /private/var/folders/q7/0kwl9rjx217dr09ps4mkc4640000gn/T/jogamp_0000/file_cache/jln12449288597341837895/jln1690886216644344164/natives/macosx-universal/libgluegen_rt.dylib: dlopen(/private/var/folders/q7/0kwl9rjx217dr09ps4mkc4640000gn/T/jogamp_0000/file_cache/jln12449288597341837895/jln1690886216644344164/natives/macosx-universal/libgluegen_rt.dylib, 0x0001): tried: '/private/var/folders/q7/0kwl9rjx217dr09ps4mkc4640000gn/T/jogamp_0000/file_cache/jln12449288597341837895/jln1690886216644344164/natives/macosx-universal/libgluegen_rt.dylib' (mach-o file, but is an incompatible architecture (have 'x86_64', need 'arm64')), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/q7/0kwl9rjx217dr09ps4mkc4640000gn/T/jogamp_0000/file_cache/jln12449288597341837895/jln1690886216644344164/natives/macosx-universal/libgluegen_rt.dylib' (no such file), '/private/var/folders/q7/0kwl9rjx217dr09ps4mkc4640000gn/T/jogamp_0000/file_cache/jln12449288597341837895/jln1690886216644344164/natives/macosx-universal/libgluegen_rt.dylib
	at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
	at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:331)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:197)
	at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:139)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2399)
	at java.base/java.lang.Runtime.load0(Runtime.java:852)
	at java.base/java.lang.System.load(System.java:2030)
	at com.jogamp.common.jvm.JNILibLoaderBase.loadLibraryInternal(JNILibLoaderBase.java:604)
	at com.jogamp.common.jvm.JNILibLoaderBase.access$000(JNILibLoaderBase.java:64)
	at com.jogamp.common.jvm.JNILibLoaderBase$DefaultAction.loadLibrary(JNILibLoaderBase.java:107)
	at com.jogamp.common.jvm.JNILibLoaderBase.loadLibrary(JNILibLoaderBase.java:488)
	at com.jogamp.common.os.DynamicLibraryBundle$GlueJNILibLoader.loadLibrary(DynamicLibraryBundle.java:427)
	at com.jogamp.common.os.Platform$1.run(Platform.java:321)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:319)
	at com.jogamp.common.os.Platform.<clinit>(Platform.java:290)
	at com.jogamp.opengl.GLProfile.<clinit>(GLProfile.java:154)
	at jogamp.opengl.ThreadingImpl$1.run(ThreadingImpl.java:83)
	at jogamp.opengl.ThreadingImpl$1.run(ThreadingImpl.java:66)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:319)
	at jogamp.opengl.ThreadingImpl.<clinit>(ThreadingImpl.java:66)
	at com.jogamp.opengl.Threading.disableSingleThreading(Threading.java:164)
	at javax.media.j3d.JoglPipeline.initialize(JoglPipeline.java:129)
	at javax.media.j3d.Pipeline.createPipeline(Pipeline.java:92)
	at javax.media.j3d.MasterControl.loadLibraries(MasterControl.java:858)
	at javax.media.j3d.VirtualUniverse.<clinit>(VirtualUniverse.java:267)
	at me.drton.jmavsim.Visualizer3D.<init>(Visualizer3D.java:194)
	at me.drton.jmavsim.Simulator.<init>(Simulator.java:193)
	at me.drton.jmavsim.Simulator.main(Simulator.java:944)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	... 2 more
```

### To Reproduce

Steps described above...

### Expected behavior

That I can use either gazebo classic or jmavsim. At some point I did have gazebo classic working, but then I updated to latest main and it all broke.

### Flight Log

None

### Software Version

...

### Flight controller

...

### Vehicle type

None

**Comment by DonLakeFlyer** [comment]
I was able to make progress by using a Linux ARM VM and running site there in Ubuntu 22. But even there it didn't work out of the box. I needed to do this to get gazebo to not crash:
```
pip uninstall empy
pip install empy==3.3.4
```

**Comment by mengchaoheng** [comment]
the same problem

**Comment by DonLakeFlyer** [comment]
So I forgot to say this was on latest Stable. I tried main on another machine and Gazebo worked without changes for me. Not sure about jmavsim. But at least for now I have a way forward with testing. @mengchaoheng You problem is different from mine. You should create a new issue.
